iam/src/providers/TwitterFollower.ts

```diff
@@ -0,0 +1,150 @@
+// ----- Types
+import type { RequestPayload, VerifiedPayload } from "@gitcoin/passport-types";
+
+// ----- Twitters OAuth2 library
+import { deleteClient, getClient, getFollowerCount } from "../procedures/twitterOauth";
+import type { Provider, ProviderOptions } from "../types";
+
+// Perform verification on twitter access token and retrieve follower count
+async function verifyTwitterFollowers(sessionKey: string, code: string): Promise<number> {
+  const client = getClient(sessionKey);
+  const data = await getFollowerCount(client, code);
+  deleteClient(sessionKey);
+  return data.followerCount;
+}
+
+// Export a Provider to carry out OAuth and return a record object
+export class TwitterFollowerGT100Provider implements Provider {
+  // Give the provider a type so that we can select it with a payload
+  type = "TwitterFollowerGT100";
+  // Options can be set here and/or via the constructor
+  _options = {};
+
+  // construct the provider instance with supplied options
+  constructor(options: ProviderOptions = {}) {
+    this._options = { ...this._options, ...options };
+  }
+
+  // verify that the proof object contains valid === "true"
+  async verify(payload: RequestPayload): Promise<VerifiedPayload> {
+    let valid = false;
+    let followerCount = 0;
+
+    try {
+      followerCount = await verifyTwitterFollowers(payload.proofs.sessionKey, payload.proofs.code);
+    } catch (e) {
+      return { valid: false };
+    } finally {
+      valid = followerCount > 100;
+    }
+
+    return {
+      valid: valid,
+      record: {
+        followerCount: followerCount?.toString(),
+      },
+    };
+  }
+}
+
+// Export a Provider to carry out OAuth and return a record object
+export class TwitterFollowerGT500Provider implements Provider {
+  // Give the provider a type so that we can select it with a payload
+  type = "TwitterFollowerGT500";
+  // Options can be set here and/or via the constructor
+  _options = {};
+
+  // construct the provider instance with supplied options
+  constructor(options: ProviderOptions = {}) {
+    this._options = { ...this._options, ...options };
+  }
+
+  // verify that the proof object contains valid === "true"
+  async verify(payload: RequestPayload): Promise<VerifiedPayload> {
+    let valid = false;
+    let followerCount = 0;
+
+    try {
+      followerCount = await verifyTwitterFollowers(payload.proofs.sessionKey, payload.proofs.code);
+    } catch (e) {
+      return { valid: false };
+    } finally {
+      valid = followerCount > 500;
+    }
+
+    return {
+      valid: valid,
+      record: {
+        followerCount: followerCount?.toString(),
```

Comment:
We should avoid putting counters as attestation and should favour statements instead, if a user gains a new follower, they could produce this same stamp with a different hash.

Instead we should consider doing something like: followerCount: valid ? "gt500" : false